src/api/c/DependencyModule.hpp

```diff
@@ -0,0 +1,55 @@
+#pragma once
+#include <dlfcn.h>
+#include <utility>
+#include <type_traits>
+#include <vector>
+#include <string>
+
+#if(OS_WIN)
+#include <Windows.h>
+using LibHandle = HMODULE;
```

Comment:
Have you considered merging this class and symbol manager class at any point ? On second thought, it may be better to keep them separate, but just wondering if would reduce any repeat code!

Reply:
It's probably a good idea. The difference is that we loading all the symbols at once whereas the unified backend is loading them as needed. I think we can merge them but I didn't want to take that on just yet.